web/src/engine/websites/PixivComics.ts

```diff
@@ -0,0 +1,269 @@
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import { Fetch, FetchJSON, FetchRequest } from '../FetchProvider';
+import type { Priority } from '../taskpool/TaskPool';
+
+type APIMangaPage = {
+    data: {
+        magazines: { id: number } []
+    }
+}
+
+type APIManga = {
+    data: {
+        official_work: {
+            id: number,
+            name: string
+        }
+    }
+}
+
+type APIMangas = {
+    data: {
+        official_works: {
+            id: number,
+            title : string
+        }[]
+    }
+}
+
+type APIChapters = {
+    data: {
+        episodes: {
+            readable: boolean,
+            episode: {
+                id: number,
+                numbering_title: string,
+                sub_title : string
+            }
+        }[]
+    }
+}
+
+type APIPages = {
+    data: {
+        reading_episode: {
+            pages: APIPage[]
+        }
+    }
+}
+
+type APIPage = {
+    url: string,
+    key: string,
+    gridsize: number,
+    width: number,
+    height: number
+}
+
+export default class extends DecoratableMangaScraper {
+    private readonly apiURL = 'https://comic.pixiv.net/api/app/';
+
+    public constructor() {
+        super('pixivcomics', `pixivコミック`, 'https://comic.pixiv.net', Tags.Language.Japanese, Tags.Media.Manga, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https?:\/\/comic\.pixiv\.net\/works\/\d+$/.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const uri = new URL(url);
+        const request = this.prepareRequest(new URL('works/v5/' + uri.pathname.match(/\d+$/)[0], this.apiURL).href);
+        const { data } = await FetchJSON<APIManga>(request);
+        const id = data.official_work.id;
+        const title = data.official_work.name.trim();
+        return new Manga(this, provider, String(id), title);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        const uri = new URL('magazines', this.apiURL);
+        const request = this.prepareRequest(uri.href);
+        const { data } = await FetchJSON<APIMangaPage>(request);
+        const pages = data.magazines.map(item => item.id);
+        for (const page of pages) {
+            const mangas = await this.getMangasFromPage(page, provider);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+    async getMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const uri = new URL(`magazines/v2/${page}/works`, this.apiURL);
+        const request = this.prepareRequest(uri.href);
+        const { data } = await FetchJSON<APIMangas>(request);
+        return data.official_works.map(item => new Manga(this, provider, String(item.id), item.title.trim()));
```

Comment:
In this case i would prefer `.toString()` over `String()` so it throws in case the `id` is something nullish.